101-aci-dynamicsnav/azuredeploy.json

```diff
@@ -54,13 +62,14 @@
       }
     },
     "variables": {
-        "image": "[parameters('navRelease')]"
+        "image": "[parameters('navRelease')]",
+        "publicdnsname": "[concat(parameters('dnsPrefix'),'.',resourceGroup().location,'.azurecontainer.io')]"
```

Comment:
@bmoore-msft I am not sure if I understand what I need to change. ".azurecontainer.io" can't be hard coded, is that the problem?

Reply:
any part of the fqdn - use the reference() function to get the entire fqdn

Reply:
@bmoore-msft sorry, I am probably missing something obvious but if I read the documentation correctly, then I can use reference() only in the output section? In my case the container needs to know in the beginning what the FQDN will be

Reply:
you can use it in the properties object on resources as well...

What docs did you find that said outputs only?  (we should fix that)

Reply:
@bmoore-msft Sorry, I misread. It only says that typically reference() is used in the output section

So I tried to do the following

```
{
    ...
    "resources": [
      {
              "name": "[parameters('name')]",
              "type": "Microsoft.ContainerInstance/containerGroups",
              ...  
              "properties": {
                  "containers": [
                      {
                          "name": "[parameters('name')]",
                          "properties": {
                              "environmentVariables": [
                               ....
                                { 
                                  "name": "PublicDnsName",
                                  "value": "[reference(resourceId('Microsoft.ContainerInstance/containerGroups/', parameters('name'))).ipAddress.fqdn]"
                                }
...
```

However this results in an error "The template resource 'Microsoft.ContainerInstance/containerGroups/acinavcontainergroup' cannot reference itself." Is that the circular reference you were mentioning in your first comment? Any ideas how to avoid that?

Reply:
@bmoore-msft any ideas on how to get the fqdn so that I can let the container know as env param what it's fqdn is? Or is that not possible?